### ISSUE TITLE
looping sampler with pre-loaded audio files not working.

### ISSUE THREAD

**Issue by sergiogranadamoreno** [enzienaudio/heavy#252]
Hi! The attached patch implements an enveloping sampler (the contents of the audio file have been saved within the patch, inside the table). This patch starts looping immediately in Pd after activating DSP, however, it won't deliver any sound after being compiled through Heavy; as in the PD patch, I would have expected to have sound at the output after increasing the amplitude parameter. 

I tried with the js and VST64bits downloads. File size is less than 1 MB.

Any clues? Thanks!
[table-test.pd.zip](https://github.com/enzienaudio/heavy/files/1765166/table-test.pd.zip)

**Comment by diplojocus** [comment]
@sergiogranadamoreno Looks like your table data is being correctly stored when uploading to heavy. 

Can you try putting a [sig~ ] object before the two [samphold~] objects?

Heavy is a bit less forgiving about receiving control connections to signal objects than Pd is. Though I have been working recently on a fix that can resolve this particular situation.

**Comment by sergiogranadamoreno** [comment]
yeah! that worked :)

so, should we always put a [sig~] before objects that deal with audio signals?
In this same patch [phasor~] receives control connections (data-flow) with no issues... does this behaviour does not applies to generators?

thanks for your help man!

**Comment by diplojocus** [comment]
Yeah for the moment some objects requires a signal connection, funnily enough [phasor~] is ok!

[lop~]
[samphold~]
[hip~]
[bp~]
[vcf~]
... tbh most of the signal rate objects (with a `~`)

I am aiming to resolve this soon!

**Comment by sergiogranadamoreno** [comment]
It doesn't seem to be a problem to me! Looping sampler is working fine now (and so this issue is solved for me).

One last question though: this patch seems to work fine as a VST plugin. Could the @hv_param functionality (from issue  #131 ) be implemented for VST development? If so, is there any docs on how to use it?

**Comment by diplojocus** [comment]
Can you make a new ticket for that? Also I'm not sure if I understand the request, exposed input parameters via @hv_param are supported in the VST target: https://enzienaudio.com/docs/index.html#09.vst#defining-parameters

**Comment by sergiogranadamoreno** [comment]
sorry, I meant the @hv_table descriptor. I'll continue on #131 .
